### PR TITLE
[sc-342712] Pierce/update provwasm 2.4.0

### PR DESCRIPTION
**Pull request by celloman**
* Updates to the latest provwasm w/ the necessary changes that entails
[sc-342712]

### CODE REVIEW

File: src/util/prov_helpers.rs [hunk 1]
```diff
@@ -1,39 +1,279 @@
...
+/// to a target address.  Assumes the parent name is unrestricted or that the contract has access to
+/// bind a name to the parent name.
+///
+/// # Parameters
+/// * `name` The dot-qualified name to use on-chain for name binding. Ex: myname.sc.pb will generate
+/// a msg that binds "myname" to the existing parent name "sc.pb".
+/// * `bind_to_address` The bech32 address to which the name will be bound.
+/// * `restricted` If true, the name will be bound as a restricted name, preventing future name
+/// bindings from using it as a parent name.
+pub fn msg_bind_name<S1: Into<String>, S2: Into<String>>(
```
Comment (celloman) [inline src/util/prov_helpers.rs:59]: stolen from https://github.com/FigureTechnologies/funding-trading-bridge-smart-contract/blob/a92bafb4397360ac0a4febfbc8390c7a54080e84/src/util/provenance_utils.rs#L20 as it handles the parent/child name relationship nicely and replicates the old functionality that we lost from provwasm w/ the upgrade

File: src/util/prov_helpers.rs [hunk 2]
```diff
@@ -1,39 +1,279 @@
...
+
+    use super::get_all_attributes;
+
+    /// Helper function to handle returning mocked responses with pagination, where the key is a json-encoded string of the next page index
+    fn mock_paged_attributes(
+        querier: &mut MockProvenanceQuerier,
+        account: &'static str,
+        pages: Vec<Vec<Attribute>>,
+    ) {
+        querier.registered_custom_queries.insert(
```
Comment (celloman) [inline src/util/prov_helpers.rs:175]: turns out this is just how complicated it is to have dynamic handling of multiple queries to the mock querier (vs. the simplistic `QueryAttributesRequest::mock_response(` type single-response-value mock)

File: src/util/prov_helpers.rs [hunk 3]
```diff
@@ -197,8 +463,93 @@ mod tests {
         );
     }
 
-    fn get_binary_int(value: u64) -> Binary {
-        to_binary(&value).expect(&format!(
+    #[test]
+    fn msg_bind_name_creates_proper_binding_with_fully_qualified_name() {
```
Comment (celloman) [inline src/util/prov_helpers.rs:467]: also stole accompanying tests for said msg_bind_name function

**Review by hyperschwartz — DISMISSED** [review]

**Review by mtps — APPROVED** [review]

**Comment by celloman** [comment]
[sc-342712]

**Review by vwagner — APPROVED** [review]

**Pull request merged**